eng/icu.proj

```diff
@@ -37,4 +37,22 @@
             Lines="@(_ItemsFromEmscriptenVersionFile->Replace('&quot;', ''))"
             Overwrite="true" />
   </Target>
+  <Target Name="PatchWasiSdk" Condition="'$(TargetOS)' == 'Wasi'">
+    <ReadLinesFromFile File="../.devcontainer/wasi-sdk-version.txt">
+    <Output TaskParameter="Lines" ItemName="_VersionLines" />
+    </ReadLinesFromFile>
+
+    <PropertyGroup>
+      <WASI_SDK_PATH>$(RepoRoot)\artifacts</WASI_SDK_PATH>
+      <WasiLocalPath>$(RepoRoot)\artifacts</WasiLocalPath>
+      <WasiSdkVersion>%(_VersionLines.Identity)</WasiSdkVersion>
+      <WasiSdkUrl>https://github.com/WebAssembly/wasi-sdk/releases/download/wasi-sdk-$(WasiSdkVersion)/wasi-sdk-$(WasiSdkVersion).0-linux.tar.gz</WasiSdkUrl>
+      <WasiSdkUrl Condition="$([MSBuild]::IsOSPlatform('OSX'))" >https://github.com/WebAssembly/wasi-sdk/releases/download/wasi-sdk-$(WasiSdkVersion)/wasi-sdk-$(WasiSdkVersion).0-macos.tar.gz</WasiSdkUrl>
+    </PropertyGroup>
+
+    <Exec Command="rm -rf wasi-sdk-*/ wasi-sdk/ &amp;&amp; curl -L -o- $(WasiSdkUrl) | tar -xvz &amp;&amp; mv wasi-sdk-$(WasiSdkVersion).*/ wasi-sdk/"
```

Comment:
This won't show any errors with downloading the tarball. Add `--fail -S` to the `curl` command.

Reply:
I guess we could do the same improvements on runtime repo